### DISCUSSION TITLE
client fetch error: csrf TypeError during email verification -- NEXTAUTH_URL is defined

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#2035] [Help]
I was using email verification successfully with an earlier version of Next-Auth 3. I updated to 3.23.1 this morning and email verification is no longer working for me. I'm getting a client fetch error csrf TypeError: failed to fetch.

NEXTAUTH_URL env var is set in Vercel to the canonical URL for my site (https://gibbs-photography.com). I deleted next-auth and my yarn file and reinstalled, as I read another thread re issues with updating canary-based installs.

I'm not seeing any errors in my Vercel function logs.

I'm calling `signIn('email', { email: email });` which, according to the docs, should fetch and attach the csrf token for me and I do see a __Host-next-auth.csrf-token cookie that is HttpOnly, Secure with a SameSite Lax policy.

I'm using Typeorm with a postgres database and TypeScript. I tried installing the legacy adapater, but that didn't seem to help matters, so, I removed it.

I'm assuming a vanilla next-auth install of 3.23.1 with a properly-configured database and the `signin(`email`, {email: email })` function should fetch the csrf token and send the verification email. But that is not the case.

Can someone see what's wrong here or suggest how I might go about tracking down the problem?

Thanks.

Brad

**Comment by balazsorban44** [comment]
A reproduction would be very helpful!

**Reply by ghost** [reply]
I'm happy to create a reproduction but thought I'd make sure it wasn't a misunderstanding on my part or some step in the upgrade process I was missing. This repository and the docs are changing pretty rapidly right now and I'm not confident there isn't something simple I missed.

**Reply by balazsorban44** [reply]
Yeah... :sweat_smile: We have been working hard, but worked hard not to make any (intentional) breaking changes though! To increase our confidence, we started working on writing some tests, and I plan to rewrite the codebase to TS pretty soon, to catch more bugs, even before they are released!

Reproduction in any case would be very helpful, and I am happy if you open it as a bug report, rather than in this discussion. Real bugs need the most attention, and I want it to be a priority, especially if you think there was a breaking change since a previous version. Could you try to pin down the version that introduced your problem? That would also help a lot. Please give it as a part of the bug report! :pray:
